### PR TITLE
Standardize scss import practices

### PR DESCRIPTION
**Pull request by brad-decker**
1. Rename the base 'index' scss files to be more descriptive
2. separates ui/app component imports into their own files
3. moves imports to their proper new location
4. Alphabetizes imports in root files.

### CODE REVIEW

**Comment by tmashuang** [comment]
Seems that some of the ordering mattered in a couple of places. A couple of the ones I found

<details>
<summary>Account Details Button Width</summary>


`.hide-token-confirmation__button, .button`  `width` and `padding` over  `.account-details-modal__button` `width` `padding`
</details>

<details>
<summary>Tx List Item tx category color</summary>


`.list-item` `color` over `.transaction-list-item--unconfirmed' `color`
</details>

<details>
<summary>Tx Details Speed Up and Cancel font size</summary>


 `.hide-token-confirmation__button, .button` `font-size` over `.transaction-list-item-details__header-button` `font-size`
</details>

<details>
<summary>Send Token button on asset list item</summary>


`button.scss` `font-size` over `.asset-list-item__send-token-button` `font-size`
 `.hide-token-confirmation__button, .button` `display` and `width` over `.asset-list-item__send-token-button` `display` and `width`

I think these are from the old design and can be removed?
</details>

<details>
<summary>Buy button width</summary>


`.hide-token-confirmation__button, .button` `width` over `.eth-overview__button, .token-overview__button` `width`
</details>

Would we just tack an `!important` for the individual cases?

**Comment by brad-decker** [comment]
@tmashuang thank you for finding those! I'll go through and review why the import order caused this to break as it likely means we are relying on styles from unrelated files which most likely is causing more heartaches than fixing things.

**Comment by brad-decker** [comment]
@tmashuang - we can avoid `!important;` by making our css selectors more specific, in this case .button and .hide-token-confirmation__button are equally specific they are both just a single class name. It's important to note that the way we are using BEM is actually making it so that our "nested" scss isn't really creating nested selectors.

```scss
.some-class {
  .child {
     font-size: 16px;
  }
}
```

compiles to
```css
.some-class .child {
  font-size: 16px;
}
```

whereas
```scss
.some-class {
  &__child {
    font-size: 16px;
  }
}
```

compiles to
```css
.some-class__child {
  font-size: 16px;
}
```

In some cases here I am adding an additional ampersand to get the nested effect
```scss
.some-class {
 & &__child {
  font-size: 16px;
 }
}
```

compiles to
```css
.some-class .some-class__child {
  font-size: 16px;
}
```
as we would hope.

**Comment by brad-decker** [comment]
I added a commit that resolves those specific issues. I'm looking through the extension now to see if i can find anymore. I did end up pulling out the mixins created ad-hoc in the buttons scss file and converted everywhere they were being used to consume the button component instead. These will likely need to be reviewed carefully for breaking changes

File: ui/app/components/ui/page-container/page-container-header/page-container-header.component.js [hunk 1]
```diff
@@ -72,7 +73,7 @@ export default class PageContainerHeader extends Component {
 
         {
           onClose && headerCloseText
-            ? <div className="page-container__header-close-text" onClick={() => onClose()}>{ headerCloseText }</div>
+            ? <Button type="link" className="page-container__header-close-text" onClick={() => onClose()}>{ headerCloseText }</Button>
```
Comment (tmashuang) [inline ui/app/components/ui/page-container/page-container-header/page-container-header.component.js:76]: Seems this changed the Close to move to a new line and to overlaid the Basic and Advanced Gas tabs, which Selenium has a failed on. 

<details>
<summary>Customize Gas Close modal text</summary>

<details>

Comment (brad-decker) [reply ui/app/components/ui/page-container/page-container-header/page-container-header.component.js:76]: Good find, I fixed this issue for the choose recipients view on the send page, but missed it here. I'll apply the same fix there and get a revision up for you.

**Comment by brad-decker** [comment]
Ready for another look @tmashuang. Thank you 🙏

**Comment by brad-decker** [comment]
As it turns out, not nesting CSS seems to be the philosophy of BEM. I can achieve the same specificity by changing `& &__element` to `&__element .sub-block`. Either or is fine for this work, but I'll be researching BEM and how we might codify our usage of it.

File: ui/app/pages/pages.scss [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
 @import 'first-time-flow/index';
+@import 'home/index';
 @import 'keychains/index';
-@import 'confirm-approve/index';
 @import 'permissions-connect/index';
-@import 'asset/asset';
+@import 'send/send';
+@import 'settings/index';
 @import 'token/index';
 @import 'create-account/index';
```
Comment (tmashuang) [inline ui/app/pages/pages.scss:20]:
```suggestion
@import 'create-account/index';
@import 'unlock-page/index;'
```

File: ui/app/pages/settings/info-tab/info-tab.component.js [hunk 3]
```diff
@@ -19,71 +20,71 @@ export default class InfoTab extends PureComponent {
...
           >
-            <span className="info-tab__link-text">
-              { t('contactUs') }
-            </span>
-          </a>
+            { t('contactUs') }
+          </Button>
         </div>
       </div>
     )
```
Comment (tmashuang) [inline ui/app/pages/settings/info-tab/info-tab.component.js:90]: Above changes have centered and padded the About page's Links section

<details>
<summary>About page's link section</summary>

</details>

Comment (brad-decker) [reply ui/app/pages/settings/info-tab/info-tab.component.js:90]: On it.

File: ui/app/pages/pages.scss (hunk 2 above)
Comment (brad-decker) [reply ui/app/pages/pages.scss:20]: whoops, rebase error. Thanks

File: ui/app/pages/settings/info-tab/info-tab.component.js (hunk 3 above)
Comment (brad-decker) [reply ui/app/pages/settings/info-tab/info-tab.component.js:90]: <details>
<summary>Fixed</summary>



</details>

File: ui/app/pages/pages.scss (hunk 2 above)
Comment (brad-decker) [reply ui/app/pages/pages.scss]: Fixed, and put in the proper order.

**Review by tmashuang — APPROVED** [review]
LGTM

**Pull request merged**